src/plugins/intel_cpu/src/shape_inference/shape_inference.cpp

```diff
@@ -355,6 +380,48 @@ class ShapeInferPaddingTA<TOp, EMPTY_PORT_MASK> : public ShapeInferPaddingBase {
     }
 };
 
+/**
+ * @brief Shape inference for v14 MaxPool.
+ *
+ * It requires dedicated port mask for CPU output shape data dependency but is not used by inference function.
+ * Review shape_infer function to include this dependency.
+ */
+template <>
+class ShapeInferPaddingTA<ov::op::v14::MaxPool, EMPTY_PORT_MASK> : public ShapeInferPaddingBase {
+public:
+    using ShapeInferPaddingBase::ShapeInferPaddingBase;
+
+    ov::optional<std::vector<StaticShape>> infer(const std::vector<StaticShapeRef>& input_shapes,
+                                                 const ov::ITensorAccessor&) override {
+        return {shape_infer(static_cast<ov::op::v14::MaxPool*>(m_node.get()), input_shapes, m_pads_begin, m_pads_end)};
+    }
+
+    port_mask_t get_port_mask() const override {
+        return util::bit::mask(0);
+    }
+};
+
+/**
+ * @brief Shape inference for v8 MaxPool.
+ *
+ * It requires dedicated port mask for CPU output shape data dependency but is not used by inference function.
+ * Review shape_infer function to include this dependency.
+ */
+template <>
+class ShapeInferPaddingTA<ov::op::v8::MaxPool, EMPTY_PORT_MASK> : public ShapeInferPaddingBase {
+public:
+    using ShapeInferPaddingBase::ShapeInferPaddingBase;
+
+    ov::optional<std::vector<StaticShape>> infer(const std::vector<StaticShapeRef>& input_shapes,
+                                                 const ov::ITensorAccessor&) override {
+        return {shape_infer(static_cast<ov::op::v8::MaxPool*>(m_node.get()), input_shapes, m_pads_begin, m_pads_end)};
+    }
+
+    port_mask_t get_port_mask() const override {
+        return util::bit::mask(0);
+    }
+};
+
```

Comment:
`Pooling::Pooling ` CPU node has been using `NgraphShapeInferFactory` without any additional mask modifications. And it seems that the Pooling node doesn't require input data dependency for shape inference. Such change may negatively impact performance, as the artificial data dependency introduce additional synchronizations to inference.

Reply:
There is no issue when `pooling` is created as `Pooling` node. 
The issue is visible if is not possible to make as `Pooling` and the `Reference` node, its case for some dynamic cases 
in ` tests/layer_tests/pytorch_tests/test_pooling.py::TestPooling` tests.

The previous Reference node always used FULL_MASK regardless if shape inference is defined or not (fallback path). I think this approach has masked issues. Now reference use mask defined in shape inference (if defined) or FULL_MASK for fallback path.

Then issue appears [in](https://github.com/praasz/openvino/blob/240a3d37cb59f0dc829d7c5eb014d8e5b3fd2984/src/plugins/intel_cpu/src/node.cpp#L1793) when the input(0) has dynamic shape but data at port(0) has static shape (exception thrown when get static shape). When dada dependency is set then static shape from data is taken and pooling output shape can be calculated.

These specialization (can be removed if problem solved) shows that issue could be fixed in:
- shape inference add additional data dependency (it looks like is not required now)
- for dynamic cases the data dependency should be always set for port 0 to get input static shape.

For `FakeQuantize` is similar issue and there are test failing if data dependency not set to port 0

Reply:
It's something unexpected. Probably we need to look into this, as I can't comprehend why do we need data to run Pooling via reference.
The main problem here is that setting `util::bit::mask(0);` impacts the main scenario - CPU plugin node inference. As such change brings artificial data dependency for the plugin node too, as it uses the common shape inference class. And in its turn, this data dependency brings negative performance impact as Pooling nodes becomes sync points in the graph without real reason. Sync points requires context switching from Shape inference to actual ops inference with additional overheads.